### PR TITLE
React Native scaffold split and new tooling

### PR DESCRIPTION
**Pull request by danielsousaio**
## Ticket

Relates to PLAT-12417, closes PLAT-12526 and PLAT-11696

## Changes introduced

This PR migrates the scaffold over to https://github.comcrowdbotics/react-native-scaffold.

Moves all the scripts that are only pertinent to the scaffold over there too.

## Test and review

- Demo app should still work and be generated via command
- Module authoring left untouched
- Scripts for module authoring continue working

New npx-based commands and help:
```
crowdbotics@modules $ npx .
usage: npx crowdbotics/modules <command>

Commands available:
  parse    Parse and validate your modules
  demo     Generate a local React Native and Django demo app
  add      Install a module in the demo app
  remove   Remove a module from the demo app
  create   Create a new module of a given type
  commit   Update an existing module from the demo source code
  upgrade  Upgrade your existing app's scaffold to the latest version

Parsing modules:
  npx crowdbotics/modules parse --source <path>

Parsing modules and writing to a json file:
  npx crowdbotics/modules parse --source <path> --write <path>

Create a module of a given type:
  npx crowdbotics/modules create --name <module-name> --type <all/react-native/django>

Upgrade your scaffold to the latest master:
  npx crowdbotics/modules upgrade

Upgrade to a specific version (git tag, git commit or branch name):
  npx crowdbotics/modules upgrade --version 2.3.0
```

### CODE REVIEW

**Review by hugo-cb — APPROVED** [review]
`npx --yes crowdbotics/modules#rn-scaffold-split parse --source modules --write my-modules.json` working perfectly!!

I did not test other commands. I suggest considering my approval under the context that I am aware of.

File: utils.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+export const invalid = (...args) => {
+  console.error(INVALID_MARK, ...args);
+  process.exit(1);
+};
+
+export const warn = (...args) => {
+  console.log(WARNING_MARK, ...args);
+};
+
+export const section = (...args) => {
```
Comment (jacunacb) [inline utils.js:20]: now that i see this function maybe instead of adding the console log that i mentioned above , you could call this function.

**Review by jacunacb — COMMENTED** [review]
@danielsousaio 🚀 . just a couple of comment from my side

File: scripts/commit-module.js [hunk 2]
```diff
@@ -16,88 +11,109 @@ const copy = (origin, target) => {
...
+    copy(sourceBackend, destBackend);
 
-const copyBackendModule = (module, originModuleDir, meta) => {
-  const backendeModuleName = module.replace(/-/g, "_");
-  const sourceBackend = path.join(demoDir, "backend", "modules", backendeModuleName);
-  const destBackend = path.join(originModuleDir, backendeModuleName);
-  console.log("backend copying ...\n", sourceBackend, destBackend);
-  copy(sourceBackend, destBackend);
-};
+    console.log("\n");
```
[the remark this answers is not in the document]
Comment (danielsousaio) [reply scripts/commit-module.js]: Good point, fixed in 8e442eaa

File: scripts/commit-module.js [hunk 3]
```diff
@@ -16,88 +11,109 @@ const copy = (origin, target) => {
...
-  copy(sourceFrontend, destFrontend);
-};
+  const copyFrontendModule = (module, originModuleDir, meta) => {
+    const sourceFrontend = path.join(demoDir, meta.root);
+    const destFrontend = path.join(originModuleDir);
+    console.log("frontend copying ...\n", sourceFrontend, destFrontend);
+    copy(sourceFrontend, destFrontend);
+  };
+
+  const copyScreenModule = (module, originModuleDir, meta) => {
```
[the remark this answers is not in the document]
Comment (danielsousaio) [reply scripts/commit-module.js]: Yes, removed it

File: utils.js (hunk 1 above)
Comment (danielsousaio) [reply utils.js:20]: Yep

File: package.json [hunk 4]
```diff
@@ -47,11 +38,12 @@
     "@babel/types": "^7.18.8",
     "ajv": "^8.11.0",
     "ajv-formats": "^2.1.1",
+    "arg": "^5.0.2",
```
[the remark this answers is not in the document]
Comment (danielsousaio) [reply package.json:41]: yes, it's used to parse arguments for the various commands, i.e.:
https://github.com/crowdbotics/modules/blob/rn-scaffold-split/index.js#L58-L61

**Comment by danielsousaio** [comment]
Could you please review this when you have a moment @jacunacb @taylor-cb ?

**Review by jacunacb — APPROVED** [review]

**Review by taylor-cb — APPROVED** [review]

**Review by jacunacb — APPROVED** [review]

**Pull request merged**